data-prepper-plugins/ml-inference-processor/src/main/java/org/opensearch/dataprepper/plugins/ml/processor/MLProcessorConfig.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.ml.processor;
+
+import com.fasterxml.jackson.annotation.JsonClassDescription;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import org.opensearch.dataprepper.model.annotations.ExampleValues;
+import org.opensearch.dataprepper.model.event.EventKey;
+import org.opensearch.dataprepper.model.event.EventKeyConfiguration;
+import org.opensearch.dataprepper.model.event.EventKeyFactory;
+import org.opensearch.dataprepper.plugins.ml.processor.configuration.ActionType;
+import org.opensearch.dataprepper.plugins.ml.processor.configuration.AwsAuthenticationOptions;
+import org.opensearch.dataprepper.plugins.ml.processor.configuration.ServiceName;
+
+@Getter
+@JsonPropertyOrder
+@JsonClassDescription("The <code>ml</code> processor enables invocation of the ml-commons plugin in OpenSearch service within your pipeline in order to process events. " +
+        "It supports both synchronous and asynchronous invocations based on your use case.")
+public class MLProcessorConfig {
+
+    @JsonProperty("aws")
+    @NotNull
```

Comment:
Does this need to be required? Even though you use `AwsCredentialsSupplier` which can use the default pipeline role and region?

Reply:
I think your point is valid. But I am following the same setup that is defined in the aws-lambda processor for consistency. https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/common/config/LambdaCommonConfig.java#L28. You can see it's defined as NotNull.